src/http-client.js

```diff
@@ -34,10 +34,27 @@ export class HttpClient {
     this.requestProcessorFactories.set(HttpRequestMessage, createHttpRequestMessageProcessor);
     this.requestProcessorFactories.set(JSONPRequestMessage, createJSONPRequestMessageProcessor);
     this.pendingRequests = [];
+    this.interceptors = [];
     this.isRequesting = false;
   }
 
   /**
+   * Add new interceptor
+   *
+   * NOTE: Interceptors are stored in reverse order. Inner interceptors before outer interceptors.
```

Comment:
Actually, is order significant? It doesn't look like interceptors get a reference to inner interceptors, and they don't appear to have a mechanism to short-circuit inner interceptors.